docs/source/55_api.md

```diff
@@ -339,6 +341,138 @@ val bcastFeatures = sc.loadFeatures("my/features.adam").broadcast()
+val variantsByFeatureShuffle = features.shuffleRegionJoinAndGroupByLeft(variants)
+
+// As a BroadcastRegionJoin, it can be implemented as follows:
+val variantsByFeatureBcast = variants.broadcastRegionJoinAndGroupByRight(features)
+```
+
+Notice that the type of join determines which dataset calls the join and which
+gets passed as an argument. In the ShuffleRegionJoin, `features` calls the
+join, but in BroadcastRegionJoin, `variants` calls the join.
+
+This distinction is very important to understanding the difference between
+BroadcastRegionJoin and ShuffleRegionJoin. It is important to first understand
+the difference between what is called the "left" and "right" datasets. Simply
+put, the left dataset is the dataset that calls the join, and the right dataset
+is what gets passed to the join.
+
+ShuffleRegionJoin Example: `leftDataset.shuffleRegionJoin(rightDataset)`
+
+BroadcastRegionJoin Example: `leftDataset.broadcastRegionJoin(rightDataset)`
+
+To perform a `groupBy` after the join, BroadcastRegionJoin only supports
+grouping by the right dataset, and ShuffleRegionJoin supports only grouping by
+the left dataset. Thus, depending on the type of join, the left and right
+dataset may change between BroadcastRegionJoin and ShuffleRegionJoin.
+
+###### Separate reads into overlapping and non-overlapping features
+
+This query joins an RDD of Reads with an RDD of features using an outer join.
+The outer join will produce an RDD where each read is optionally mapped to a
+feature. If a given read does not overlap with any features provided, it is
+paired with a `None`. After we perform the join, we use a predicate to separate
+the reads into two RDDs. This query is useful for filtering out reads based on
+feature data. For example, identifying reads that overlap with ChIPSeq data to
+perform chromatin accessibility studies. It may be useful to separate the reads
+to perform distinct analyses on each resulting dataset.
+
+```scala
+// An outer join provides us with both overlapping and non-overlapping data
+val reads = sc.loadAlignments(“my/reads.adam”)
+val features = sc.loadFeatures(“my/features.adam”)
+
+// As a ShuffleRegionJoin, we can use a LeftOuterShuffleRegionJoin:
+val readsToFeatures = reads.leftOuterShuffleRegionJoin(features)
+
+// As a BroadcastRegionJoin, we can use a RightOuterBroadcastRegionJoin:
+val featuresToReads = features.rightOuterShuffleRegionJoin(reads)
+
+// After we have our join, we need to separate the RDD
+// If we used the ShuffleRegionJoin, we filter by None in the values
+val overlapsFeatures = readsToFeatures.rdd.filter(_._2 != None)
+val notOverlapsFeatures = readsToFeatures.rdd.filter(_._2 == None)
+
+// If we used BroadcastRegionJoin, we filter by None in the keys
+val overlapsFeatures = featuresToReads.rdd.filter(_._1 != None)
+val notOverlapsFeatures = featuresToReads.rdd.filter(_._1 != None)
+```
+
+Previously, we illustrated that join calls can be different between
```

Comment:
The Previously, and We also previously paragraphs.  I don't feel strongly about it.